org-roam-capture.el

```diff
@@ -277,25 +277,54 @@ This function is used solely in Org-roam's capture templates: see
     (widen)
     (goto-char (point-max))))
 
+(define-error 'org-roam-malformed-template
+  "Malformed template in `org-roam-capture-templates'"
+  'org-roam-errors)
+
+(defun org-roam-capture--validate-template (template)
+  "Validate TEMPLATE by checking its elements.
+Return t if the TEMPLATE is well-formed, nil otherwise.
+As a special case, return 'group if TEMPLATE is a template-group.
```

Comment:
Maybe that behaviour would warrant another name than `validate`, but since `'group` is still `t`, I think it works.